src/providers/symbols.ts

```diff
@@ -0,0 +1,12 @@
+import * as vscode from 'vscode';
+import lsp from '../lsp/main';
+import * as util from '../utilities';
+
+export class CalvaDocumentSymbolsProvider implements vscode.DocumentSymbolProvider {
+  provideDocumentSymbols(
+    document: vscode.TextDocument,
+    _token: vscode.CancellationToken
+  ): vscode.ProviderResult<vscode.SymbolInformation[] | vscode.DocumentSymbol[]> {
+    return lsp.getDocumentSymbols(util.cljsLib.getStateValue(lsp.LSP_CLIENT_KEY), document.uri);
```

Comment:
This returns a promise, and from what I see in the debugger decorations code, once the promise is resolved, you may need to take the `children` of the first item of the resulting array.

https://github.com/BetterThanTomorrow/calva/blob/25f18d2821a7846a9045d0ab0ab4581e610a73d1/src/debugger/decorations.ts#L50

Reply:
So maybe something like this would work:

Make `provideDocumentSymbols` async, then do this:

```typescript
const symbolsResult = await lsp.getDocumentSymbols(util.cljsLib.getStateValue(lsp.LSP_CLIENT_KEY), document.uri);
return symbolsResult[0].children;
```

Reply:
I don't remember the structure or the details of the response from clojure-lsp, but if it's the case that we always want the children of the first item in the array, maybe we should move that into the `getDocumentSymbols` function.

Reply:
I have actually tried what you suggest there, and confirmed that the symbols to get assigned to `symbolsResult`, and also tried to return only the children. However, I don't think that this with `children` is the culprit. clojure-lsp creates the symbols as children to the namespace symbol, and in the case of the debugger we are only interested in these children. The problem I have with this PR is that _nothing_ gets provided.

Reply:
Does the disabling of the middleware cause this, or did you verify that it's not called when you manually make the request?

Reply:
Not sure I follow. But this provider is being called, and the symbols are being gathered. They are not being provided, though. (The Outline remains un-populated.)

Reply:
I think I'll close this PR. It is a very limited use case and having a separate group of symbols can be good enough.

Reply:
I thought you meant `lsp.getDocumentSymbols` was not returning symbols. And I was wondering if the middleware being changed to return `null` was the cause, but it seems you're saying that you were getting symbols from lsp, just that after returning